service/history/replicationTaskFetcher.go

```diff
@@ -284,7 +299,9 @@ func (f *ReplicationTaskFetcherImpl) getMessages(
 	}
 	response, err := f.remotePeer.GetReplicationMessages(ctx, request)
 	if err != nil {
-		return nil, err
+		if _, ok := err.(*serviceerror.ResourceExhausted); !ok {
+			return nil, err
+		}
 	}
 
 	return response.GetShardMessages(), err
```

Comment:
this logic is weird, i.e.
if err is not nil and of type `ResourceExhausted` -> `response.GetShardMessages()` causing NPE?

i guess the previous logic is good for now
```go
if err != nil {
	return nil, err
}
```

Reply:
actually, turns out the proto layer already handles the nil case.

i think it is still preferable to do the following```go
if err != nil {
	return nil, err
}
```
to make the code less confusing

Reply:
I don't see where it handles `nil` case. If `err` is not `nil` response is always `nil`. I am removing this change.